src/Hosting.CommandLine/HostBuilderExtensions.cs

```diff
@@ -31,9 +31,33 @@ public static class HostBuilderExtensions
         /// <param name="cancellationToken">A cancellation token</param>
         /// <returns>A task whose result is the exit code of the application</returns>
         public static async Task<int> RunCommandLineApplicationAsync<TApp>(
-            this IHostBuilder hostBuilder, string[] args, CancellationToken cancellationToken = default)
+            this IHostBuilder hostBuilder,
+            string[] args,
+            CancellationToken cancellationToken = default)
+            where TApp : class
+        {
+            return await RunCommandLineApplicationAsync<TApp>(hostBuilder, args, null, cancellationToken);
+        }
+
+        /// <summary>
+        ///     Runs an instance of <typeparamref name="TApp" /> using <see cref="CommandLineApplication" /> to provide
+        ///     command line parsing on the given <paramref name="args" />.  This method should be the primary approach
+        ///     taken for command line applications.
+        /// </summary>
+        /// <typeparam name="TApp">The type of the command line application implementation</typeparam>
+        /// <param name="hostBuilder">This instance</param>
+        /// <param name="args">The command line arguments</param>
+        /// <param name="configure">The delegate to configure the application</param>
+        /// <param name="cancellationToken">A cancellation token</param>
+        /// <returns>A task whose result is the exit code of the application</returns>
+        public static async Task<int> RunCommandLineApplicationAsync<TApp>(
+            this IHostBuilder hostBuilder,
+            string[] args,
+            Action<CommandLineApplication> configure,
```

Comment:
```suggestion
            Action<CommandLineApplication<TApp>> configure,
```
If we make this use the generic parameter, then users can also access the `Model` and `ModelFactory` properties.